src/mono/mono/utils/mono-cgroup.c

```diff
@@ -0,0 +1,972 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+/*++
+
+Module Name:
+
+    mono-cgroup.c
+
+Abstract:
+    Read the memory limit for the current process
+
+    Adapted from runtime src/coreclr/gc/unix/cgroup.cpp 
```

Comment:
Can we make a common version of this in `src/native` and share between the two runtimes?

Reply:
The coreCLR code is C++. So far as I can tell mono is entirely C. Is this something that can be done down the track? I am unsure how straightforward things would be.

Reply:
@nealef yea we woudl need to port the CoreCLR version to C (already done) and make C++ wrappers.  Similar to what was done for EventPipe.   I dont' think it needs to be part of this PR.  But we should make an issue to re-unify the code.